packages/flutter/test/widgets/sensitive_content_host_test.dart

```diff
@@ -0,0 +1,2271 @@
+  // Default content sensitivity setting for testing.
+  const ContentSensitivity defaultContentSensitivitySetting = ContentSensitivity.autoSensitive;
+
+  // The state of content sensitivity in the app.
+  final SensitiveContentHost sensitiveContentHost = SensitiveContentHost.instance;
+
+  // The ContentSenstivity levels that get set by the native platform via calls to
+  // `SensitiveContent.setContentSensitivity`.
+  List<ContentSensitivity> setContentSensitivityArgs = <ContentSensitivity>[];
+
+  setUp(() {
+    // Reset number of method channel calls to `SensitiveContent.setContentSensitivity`.
+    setContentSensitivityArgs = <ContentSensitivity>[];
+
+    // Mock calls to the sensitive content method channel.
+    TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+      SystemChannels.sensitiveContent,
+      (MethodCall methodCall) async {
+        if (methodCall.method == 'SensitiveContent.setContentSensitivity') {
+          setContentSensitivityArgs.add(
+            ContentSensitivity.values.firstWhere(
+              (ContentSensitivity cs) => cs.name == methodCall.arguments as String,
+            ),
+          );
+        } else if (methodCall.method == 'SensitiveContent.getContentSensitivity') {
+          return defaultContentSensitivitySetting.name;
+        } else if (methodCall.method == 'SensitiveContent.isSupported') {
+          return true;
+        }
+        return null;
+      },
+    );
+  });
+
+  tearDown(() {
+    TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+      SystemChannels.sensitiveContent,
+      null,
+    );
+  });
+
+  testWidgets('one SenstiveContent widget sets content sensitivity for tree as expected', (
+    WidgetTester tester,
+  ) async {
+    await tester.pumpWidget(
+      SensitiveContent(sensitivity: ContentSensitivity.sensitive, child: Container()),
+    );
+
+    expect(sensitiveContentHost.calculatedContentSensitivity, equals(ContentSensitivity.sensitive));
+    expect(setContentSensitivityArgs, <ContentSensitivity>[ContentSensitivity.sensitive]);
+  });
+
+  testWidgets(
+    'disposing only SensitiveContent widget in the tree sets content sensitivity back to the default as expected',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        SensitiveContent(sensitivity: ContentSensitivity.sensitive, child: Container()),
+      );
+      await tester.pumpWidget(Container());
+
+      expect(sensitiveContentHost.calculatedContentSensitivity, isNull);
+      expect(setContentSensitivityArgs, <ContentSensitivity>[
+        ContentSensitivity.sensitive,
+        defaultContentSensitivitySetting,
+      ]);
+    },
+  );
+
+  group(
+    'one sensitive SensitiveContent widget in the tree determines content sensitivity for tree as expected',
+    () {
+      // Tests with other sensitive widget(s):
+      testWidgets('with another sensitive widget ca', (WidgetTester tester) async {
```

Comment:
What is `ca`?

Reply:
Accident. There are some others like this, as well. Removed!